### ISSUE TITLE
Empty characters in ligature font

### ISSUE THREAD

**Issue by tijmenvangurp** [fabiospampinato/icon-font-buildr#5]
![image](https://user-images.githubusercontent.com/9990026/57009295-02866600-6bf6-11e9-963c-d13d38a72d57.png)

I get that in order to use ligatures you need the characters to be in the font. The only problem is they are created as empty glyphs

These are the ligatures used `debug,   debug_alt,  backup, bug_report, amazon`

Not having any icons in those places gives a few challenges

Searching the font in our design tool Sketch, uses the characters of the font for the preview. them being empty causes a problem.
![image](https://user-images.githubusercontent.com/9990026/57009402-aa039880-6bf6-11e9-9db9-a5e4da23e179.png)
_As you can see here you only see the capital letter M in the preview_

Replacing the icon by typing is like typing in the blind because no characters appear. 

I tried to solve it by actually adding all the characters as icons `a,b,c,...` but this gives an issue, basically your script says there are duplicated codepoints (which I guess there are).

Long story short: would be nice if the script allows for adding your own glyphs for characters a-z

**Comment by tijmenvangurp** [comment]
Maybe a bit more helpful information on what I tried

With the following extra file
[a.svg.zip](https://github.com/fabiospampinato/icon-font-buildr/files/3134461/a.svg.zip)

and the following object in my icons:

```
{       
        "name": "a",
        "codepoints": ["\u0061"],
        "icon": "a"
}
```
Gives the folowing error:
`UnhandledPromiseRejectionWarning: Error: Given codepoints for the glyph "a" contain duplicates.`

**Comment by fabiospampinato** [comment]
I'm not familiar with the tools you're using. Help me better understand the problem, you'd like icons to be replacing the a-z unicode interval and this tool isn't letting you do that?

**Comment by tijmenvangurp** [comment]
Yes, I am trying to replace the `empty` a-z interface because my design software tries to create a preview of the font, with the actual font. Because it is missing those characters it is showing nothing.

**Comment by tijmenvangurp** [comment]
I think this has nothing to do with your code after some more investigation

This is the origin of the error
https://github.com/nfroidure/svgicons2svgfont/blob/master/src/index.js#L189 

Wonder if it is even possible to add icons on the ligature places, or that they actually are needed as empty glyphs. Do you have any thoughts on this?

Would not mind if you close this issue since it does not relate to your code. Would appreciate it if you had any input for me.

**Comment by tijmenvangurp** [comment]
I think I found the issue in your code.

I added 2 logs
```
   if ( this.config.output.codepoints ) unicode.push ( ...codepoints );
    if ( this.config.output.ligatures ) unicode.push ( ...ligatures );
    console.log(name);
        console.log(unicode);
```

Which outputs:

```
[ 'a', 'a' ]
(node:53346) UnhandledPromiseRejectionWarning: Error: Given codepoints for the glyph "a" contain duplicates.
    at SVGIcons2SVGFontStream._transform
```
Meaning it tries to add the character 'a' 2 times. 

Can it be if you have ligatures enabled, that it always adds a ligature even if none is defined? in this case the ligature is the same as the codepoint?

Might be good to build an exception for this case?

**Comment by fabiospampinato** [comment]
Shouldn't you just disable ligatures and/or codepoints in this situation?

**Comment by tijmenvangurp** [comment]
You can close the issue.

I was able to add the characters 

The only thing needed was a ligature that is different than the character.  Since I wanted to use ligatures for the other icons but not per se for this one I left out the ligature statement, assuming it would not set any ligature. I was not aware it would take the name as ligature when there was no ligature defined. 
```
{       
        "name": "a",
        "codepoints": ["\u0061"],
        "icon": "a",
        "ligatures": ["char-a"]

}
```
